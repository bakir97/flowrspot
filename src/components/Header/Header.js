import React, { Component } from "react";
//Components
import Logo from "./Logo";
import NavLinks from "./NavLinks";
import MobileNav from "./MobileNav";
//
import styles from "./Header.module.scss";
import loggedUser from "./helperFunctions/loggedUser";
import { clientWidth } from "./helperFunctions/utils";
export default class Header extends Component {
  state = {
    width: clientWidth(),
    showMenu: false
  };

  toggleMenu = () => {
    this.setState(prevState => ({ showMenu: !prevState.showMenu }));
  };
  componentWillMount() {
    window.onresize = this.resizeFunction;
  }
  resizeFunction = () => {
    this.setState({
      width: clientWidth()
    });
  };
  render() {
    const { width, showMenu } = this.state;
    const { user, openModal } = this.props;

    const isMobile = width <= 900 ? true : false;
    return (
      <>
        <div className={styles.container}>
          <Logo />

          {isMobile ? (
            <MobileNav toggleMenu={this.toggleMenu} />
          ) : (
            <>
              <NavLinks closeModal={this.props.closeModal} />
              {loggedUser(user, openModal, showMenu, this.toggleMenu)}
            </>
          )}
        </div>
        {showMenu && (
          <div className={styles.mobile_container}>
            <NavLinks
              showMenu={showMenu}
              closeModal={this.props.closeModal}
              toggleMenu={this.toggleMenu}
            />
            {loggedUser(user, openModal, showMenu, this.toggleMenu)}
          </div>
        )}
      </>
    );
  }
}
