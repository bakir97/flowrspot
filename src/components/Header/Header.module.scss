.container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 20px;
  height: 80px;
}
