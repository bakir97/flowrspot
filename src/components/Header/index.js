import Header from "./Header";
import { connect } from "react-redux";
import { openModal, closeModal } from "../../redux/modal/actions";
import { getModalType } from "../../redux/modal/selectors";
import { getUser } from "../../redux/user/selectors";
const mapStateToProps = state => ({
  modal: getModalType(state),
  user: getUser(state)
});

const mapDispatchToProps = {
  openModal,
  closeModal
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
