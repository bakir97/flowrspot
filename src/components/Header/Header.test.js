import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
describe("Header", () => {
  it("should render correctly", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
  it("should render MobileNav", () => {
    const component = shallow(<Header />);
    component.setState({ width: 700 });
    expect(component).toMatchSnapshot();
  });
  it("should show mobile menu", () => {
    const component = shallow(<Header />);
    component.setState({ width: 700, showMenu: true });
    expect(component).toMatchSnapshot();
  });
});
