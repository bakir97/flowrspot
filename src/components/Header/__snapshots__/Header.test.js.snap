// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render MobileNav 1`] = `
<Fragment>
  <div
    className="container"
  >
    <Logo />
    <MobileNav
      toggleMenu={[Function]}
    />
  </div>
</Fragment>
`;

exports[`Header should render correctly 1`] = `
<Fragment>
  <div
    className="container"
  >
    <Logo />
    <NavLinks />
    <LoginButton
      showMenu={false}
      toggleMenu={false}
    />
    <CreateAccountButton
      showMenu={false}
      toggleMenu={false}
    />
  </div>
</Fragment>
`;

exports[`Header should show mobile menu 1`] = `
<Fragment>
  <div
    className="container"
  >
    <Logo />
    <MobileNav
      toggleMenu={[Function]}
    />
  </div>
  <div
    className="mobile_container"
  >
    <NavLinks
      showMenu={true}
      toggleMenu={[Function]}
    />
    <LoginButton
      showMenu={true}
      toggleMenu={[Function]}
    />
    <CreateAccountButton
      showMenu={true}
      toggleMenu={[Function]}
    />
  </div>
</Fragment>
`;
